src/tree/hist/evaluate_splits.h

```diff
@@ -412,6 +412,7 @@ class HistEvaluator {
     tree_evaluator_.AddSplit(candidate.nid, left_child, right_child,
                              tree[candidate.nid].SplitIndex(), left_weight,
                              right_weight);
+    evaluator = tree_evaluator_.GetEvaluator();
```

Comment:
Does `AddSplit()` change the evaluator object in `tree_evaluator_`?

Reply:
Yes, my oversight. It resize the storage according to number of nodes. The tests couldn't detect the error as it happens rarely due to pre-allocation and over-allocation for efficient GPU training.